packages/canary-react/src/stories/ic-data-table.stories.mdx

```diff
@@ -1378,3 +1378,137 @@ const DataTable = () => (
 
 export default DataTable;
 ```
+
+### Updating data
+
+export const UpdatingData = () => {
+  const [data, setData] = useState([]);
+  const [pagination, setPagination] = useState([{ label: '5', value: '5' }]);
+  
+  const addData = () => {
+    setData([...data, ...LONG_DATA]);
+  }
+
+  const clearData = () => {
+    setData([]);
+  }
+
+  const updatePagination = () => {
+    setPagination([
+      { label: '10', value: '10' },
+      { label: '20', value: '20' },
+      { label: '50', value: '50' }
+    ]);
+  }
+
+  const resetPagination = () => {
+  setPagination([{ label: '5', value: '5' }]);
+  }
+
+  return (
+    <>
+      <IcDataTable
+        caption="Updating Data"
+        columns={COLS}
+        data={data}
+        showPagination
+        paginationOptions={{
+          itemsPerPage: pagination,
+          itemsPerPageControl: true,
+          goToPageControl: true,
+          type: "page",
+        }}
+      />
+      <div style={{ display: "flex", gap: "8px", marginTop: "8px" }}>
+        <IcButton onClick={addData}>Add data</IcButton>
+        <IcButton onClick={updatePagination}>Update page lengths</IcButton>
+        <IcButton onClick={clearData}>Reset data</IcButton>
+        <IcButton onClick={resetPagination}>Reset page lengths</IcButton>
+      </div>
+    </>
+  );
+};
+
+<Canvas withSource="none">
+  <Story name="Updating data example">
+    <UpdatingData />
+  </Story>
+</Canvas>
+
+#### Updating data code example
+
+```jsx
+import * as React from "react";
+import { IcDataTable } from "@ukic/canary-react";
+import { IcDataTableColumnObject } from "@ukic/canary-web-components";
+
+const columns: IcDataTableColumnObject[] = [
+  {
+    key: "firstName",
+    title: "First name",
+    dataType: "string",
+  },
+  ...
+];
+
+const data = [
+  {
+    firstName: "Joe",
+    lastName: "Bloggs",
+    age: 30,
+    jobTitle: "Developer",
+    address: "1 Main Street, Town, County, Postcode",
+  },
+  ...
+];
+
+const DataTable = () => {
+  const [data, setData] = useState([]);
+  const [pagination, setPagination] = useState([{ label: '5', value: '5' }]);
+
+  const addData = () => {
+    setData([...data, ...LONG_DATA]);
+  }
+
+  const clearData = () => {
+    setData([]);
+  }
+
+  const updatePagination = () => {
+    setPagination([
+      { label: '10', value: '10' },
+      { label: '20', value: '20' },
+      { label: '50', value: '50' }
+    ]);
+  }
+
+  const resetPagination = () => {
+    setPagination([{ label: '5', value: '5' }]);
+  }
+
+ return (
+    <>
+      <IcDataTable
+        caption="Updating Data"
+        columns={COLS}
+        data={data}
+        showPagination
+        paginationOptions={{
+          itemsPerPage: pagination,
```

Comment:
Only thing missing from the previous PR feedback is the consistent variable name to the canary-web-components code snippets. It would be good to keep them the same where possible. As this relates to the state `pagination`, it might be worth updating `defaultPagOptions` on the canary-web-components code snippets to `pagination`.

Reply:
updated both to use `pageOptions` as mentioned in previous PR